app/models/episode.rb

```diff
@@ -223,10 +223,20 @@ def medium=(new_medium)
     super
 
     if medium_changed? && medium_was.present?
-      unless medium == "audio" && medium_was == "uncut"
+      if medium_was == "uncut" && medium == "audio"
+        uncut&.mark_for_replacement
+      elsif medium_was == "audio" && medium == "uncut"
+        if contents.first&.status_complete?
+          (uncut || build_uncut).tap do |u|
+            u.original_url = contents.first.url
+            u.file_size = contents.first.file_size
+            u.duration = contents.first.duration
+          end
+        end
+        contents.each(&:mark_for_replacement)
+      else
```

Comment:
Refactored to be more explicit, in how it handles uncut->audio and audio->uncut.